### PR TITLE
Bitwarden Integration

### PR DESCRIPTION
**Pull request by tomasstrba**
Task/Issue URL:  https://app.asana.com/0/72649045549333/1202684343604567/f
iOS PR:  https://github.com/duckduckgo/iOS/pull/1399
macOS PR: https://github.com/more-duckduckgo-org/macos-browser/pull/843
What kind of version bump will this require?: Major

**Optional**:

Tech Design URL:
https://app.asana.com/0/481882893211075/1202715202076385/f
https://app.asana.com/0/481882893211075/1202870974810231/f
CC: @samsymons, @GioSensation 

**Description**:

**Steps to test this PR**:
See https://github.com/more-duckduckgo-org/macos-browser/pull/843

**OS Testing**:

* [x] iOS 14
* [x] iOS 15
* [x] iOS 16
* [x] macOS 10.15
* [x] macOS 11
* [x] macOS 12

---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)

### CODE REVIEW

File: Sources/BrowserServicesKit/Autofill/AutofillUserScript.swift [hunk 1]
```diff
@@ -179,7 +185,7 @@ extension AutofillUserScript: WKScriptMessageHandlerWithReply {
                                       didReceive message: WKScriptMessage,
                                       replyHandler: @escaping (Any?, String?) -> Void) {
         guard let messageHandler = messageHandlerFor(message.name) else {
-            // Unsupported message fail silently
+            assertionFailure("Unsupported message")
```
Comment (samsymons) [inline Sources/BrowserServicesKit/Autofill/AutofillUserScript.swift:188]: Nice :+1:

File: Sources/BrowserServicesKit/Autofill/AutofillUserScript+SecureVault.swift [hunk 2]
```diff
@@ -143,8 +153,9 @@ extension AutofillUserScript {
     }
 
     struct CredentialObject: Codable {
-        let id: Int64
+        let id: String
```
Comment (THISISDINOSAUR) [inline Sources/BrowserServicesKit/Autofill/AutofillUserScript+SecureVault.swift:156]: I remember a while back we changed some of the data type casting we were doing for a good reason, but I don't remember what the reason was. Just want to check everyone on the JS side is aware of the change?

Comment (tomasstrba) [reply Sources/BrowserServicesKit/Autofill/AutofillUserScript+SecureVault.swift:156]: @THISISDINOSAUR, good point! The reason we need to change the type here is that Bitwarden ids are Strings. @GioSensation informed me, JS layer already worked with Strings even before this change.

Comment (THISISDINOSAUR) [reply Sources/BrowserServicesKit/Autofill/AutofillUserScript+SecureVault.swift:156]: Oh interesting, I assume it's fine then. tbh I don't remember what the original change was, @shakyShane do you remember?

**Review by samsymons — APPROVED** [review]
LGTM by way of testing the client PRs

**Pull request merged**